CentralComputing/Configurator.cpp

```diff
@@ -4,33 +4,32 @@ using namespace std;
 
 Configurator ConfiguratorManager::config;
 
-int Configurator::openVarFile(string fileName) {
-    inFile.open(fileName);
-    if(!inFile) {
-        return 0;
-    }
-    loadValues();
-    return 1;
+bool Configurator::openConfigFile(const string& fileName) {
+  inFile.open(fileName);
+  if(!inFile) {
+    return false;
+  }
+  loadValues();
+  inFile.close();
+  return true;
 }
 
 void Configurator::loadValues() {
-    inFile.clear();
-    inFile.seekg(0, ios::beg);
-    string varName;
-    double val;
-    int count = 0;
-    while(inFile >> varName){
-	inFile >> val;
-	cout << varName << " " << val << " " << count<< endl;
-	count++;
-	mapVals.insert(pair<string, double> (varName, val));
-    }
+  inFile.clear();
+  inFile.seekg(0, ios::beg);
+  string varName;
+  double val;
+  while(inFile >> varName){
+    inFile >> val;
+    mapVals.insert(pair<string, double> (varName, val));
+  }
 }
 
-double Configurator::getValue(string varName) {
-    if (mapVals.find(varName) == mapVals.end()) {
-	return 0.0;
-    }
-    return mapVals.find(varName)->second;
+bool Configurator::getValue(const string& varName, double& value) {
```

Comment:
I also changed the signature to include the `double& val`, which is another reference, and now this function returns a `bool`. This change allows us to determine if our target string is incorrect. Since `0` is a valid option for some variables, we don't want to accept `0` when actually we just passed in the incorrect string into this function.